satellite/metainfo/metainfo.go

```diff
@@ -363,6 +363,9 @@ func (endpoint *Endpoint) DownloadSegmentOld(ctx context.Context, req *pb.Segmen
 	} else if pointer.Type == pb.Pointer_REMOTE && pointer.Remote != nil {
 		limits, privateKey, err := endpoint.orders.CreateGetOrderLimits(ctx, bucketID, pointer)
 		if err != nil {
+			if orders.ErrDownloadFailedNotEnoughPieces.Has(err) {
+				mon.Meter(fmt.Sprintf("%s_%s", "download_failed_not_enough_pieces_uplink", req.GetSegment())).Mark(1) //locked
```

Comment:
Hmm.. so I don't think the metric locking code recognizes anything other than string literals. In other words,

```go
mon.Meter("this_will_work")
mon.Meter(fmt.Sprintf("this_%s_work", "wont"))
```

Additionally, we don't really want to have metrics where the name includes things that have unlimited cardinality. So things like the paths, stream ids, segment ids, etc. should not be part of the name. Not only would it be a memory leak as time went on (the process would never free these created metrics in memory), metrics databases scale much more poorly with the number of metrics versus the amount of data stored per metric.

So I'd suggest making these just named `"download_failed_not_enough_pieces_uplink"` or whatever, and using logs to output information about things like segment id or stream id or path.

Also, I'm pretty sure the streamID is like 40KB (!!) which is not at all obvious lol.

Reply:
Did not know that! The reason why I added the streamID and path is because the message below from @littleskunk 
``` 
we have to group by pieceID otherwise we count the same piece more than once
100 failed download doesn't mean we have 100 bad segments. It could be the same segment with 100 download attempts
```
Do you have some ideas on how we could group the metrics by ?

Reply:
Well, we could keep a `map[piece]struct{}` in memory and only mark the meter if we add a new one to it. That's not ideal in a distributed setting because multiple workers will possibly count the same piece.

Personally, I'm happy with the meter counting duplicates, and if it's ever over zero, you look at the logs to figure out the set of pieces that are busted. Metrics aren't a tool for every problem, and I think we'll have a bad time trying to fit it in here. I think we should make sure that logs are easily available and searchable and rely on that system here.